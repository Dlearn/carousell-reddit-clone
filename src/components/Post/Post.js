// src/components/Post/Post.js

import React, { Component } from 'react';
import { Button } from 'react-bootstrap';

class Post extends Component {
  render() {
    let post = this.props.posts[this.props.i];
    let numbering = parseInt(this.props.i, 10) + 1;

    return (
      <div className="jumbotron">
        <h2 id="title">{numbering}: { post.title }</h2>
        <h4 id="subtitle" className="text-muted">Submitted {post.hoursAgo} hours ago by {post.author}.</h4>
        <br />
        <h4 id="votes">Votes: { post.votes }</h4>
        <Button 
          id = "upvote-button"
          onClick= { () => this.props.handleUpvote(post) }
          bsStyle="success" 
          bsSize="small"
        >
          Upvote
        </Button>
        <Button
          id = "downvote-button"
          onClick= { () => this.props.handleDownvote(post) }
          bsStyle="danger" 
          bsSize="small"
        >
          Downvote
        </Button>
      </div>
    );
  }
}

export default Post;