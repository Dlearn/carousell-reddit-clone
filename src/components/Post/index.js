// src/components/Post/index.js

import React, { Component } from 'react';
import { Button } from 'react-bootstrap';

class Post extends Component {
  render() {
    let post = this.props.posts[this.props.i];
    let numbering = parseInt(this.props.i, 10) + 1;

    return (
      <div className="jumbotron">
        <h2>{numbering}: { post.title }</h2>
        <br />
        <Button 
          onClick= { () => this.props.handleUpvote(post) }
          bsStyle="success" 
          bsSize="small"
        >
          Upvote
        </Button>
        <div>Upvotes: { post.upvotes }</div>
        <Button
          onClick= { () => this.props.handleDownvote(post) }
          bsStyle="danger" 
          bsSize="small"
        >
          Downvote
        </Button>
        <div>Downvotes: { post.downvotes }</div>
      </div>
    );
  }
}

export default Post;