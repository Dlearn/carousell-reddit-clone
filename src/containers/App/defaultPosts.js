// src/containers/App/defaultPosts.js

let defaultPosts = [
  {
    title: "Why isn't Trump impeached yet?",
    upvotes: 100,
    downvotes: 43,
  },
  {
    title: "My Little Pony is the best show ever",
    upvotes: 90,
    downvotes: 36,
  },
  {
    title: "What is your favorite board game?",
    upvotes: 80,
    downvotes: 20,
  },
  {
    title: "The quality of this subreddit is not good...",
    upvotes: 70,
    downvotes: 20,
  },
  {
    title: "Anybody concerned we only have upvotes in multiples of 10s?",
    upvotes: 60,
    downvotes: 20,
  },
  {
    title: "Marvel or DC?",
    upvotes: 50,
    downvotes: 12,
  },
  {
    title: "What are some of your favorite books?",
    upvotes: 40,
    downvotes: 30,
  },
  {
    title: "MEEEEEEEEEMES!!",
    upvotes: 30,
    downvotes: 20,
  },
  {
    title: "How to build a jetpack?",
    upvotes: 29,
    downvotes: 20,
  },
  {
    title: "My Elsa Cosplay! Upvote!",
    upvotes: 28,
    downvotes: 20,
  },
  {
    title: "I just read Tale of two Cities. WOW.",
    upvotes: 27,
    downvotes: 20,
  },
  {
    title: "I miss scooby dooby doo...",
    upvotes: 26,
    downvotes: 20,
  },
  {
    title: "Updoot for love y'all!",
    upvotes: 25,
    downvotes: 20,
  },
  {
    title: "If you upvote this, they will see this image when they google.",
    upvotes: 24,
    downvotes: 20,
  },
  {
    title: "Stranger things have happened since...",
    upvotes: 3,
    downvotes: 20,
  },
  {
    title: "Anybody want to make friends?",
    upvotes: 2,
    downvotes: 20,
  },
  {
    title: "I'm not a good painter, but I made this in my spare time.",
    upvotes: -5,
    downvotes: 20,
  },
  {
    title: "I just donated $10000 to North Korea! Help!",
    upvotes: -10,
    downvotes: 20,
  },
  {
    title: "Everyone should watch Avatar: The Last Airbender.",
    upvotes: -30,
    downvotes: 2,
  },
  {
    title: "Manga, to read or not to read?",
    upvotes: -50,
    downvotes: 0,
  },
  {
    title: "Batman vs Superman is the best movie everr.",
    upvotes: -100,
    downvotes: 0,
  },
];

export { defaultPosts };