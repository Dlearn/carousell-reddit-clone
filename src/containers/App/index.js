// src/containers/App/index.js

import React, { Component } from 'react';

class Post extends Component {
  render() {
    let post = this.props.posts[this.props.i];
    console.log(post);
    return (
      <div key={ this.props.i }>
        <div>Title: { post.title }</div>
        <div>Upvotes: { post.upvotes }</div>
        <div>Downvotes: { post.downvotes }</div>
        <button 
          onClick= { () => this.props.handleUpvote(post) }
        >
          Upvote
        </button>
        <button
          onClick= { () => this.props.handleDownvote(post) }
        >
          Downvote
        </button>
      </div>
    );
  }
}

class App extends Component {
	constructor() {
  	super();

    this.state = {
      input: "",
      posts: [],
    };

    // Populate the list with some dummy posts.
    this.state.posts.push({
      title: "Why isn't Trump impeached yet?",
      upvotes: 100,
      downvotes: 43,
    });

    this.state.posts.push({
      title: "Everyone should watch Avatar: The Last Airbender.",
      upvotes: 5,
      downvotes: 2,
    });

    this.state.posts.push({
      title: "Manga, to read or not to read?",
      upvotes: 4,
      downvotes: 0,
    });

    this.state.posts.push({
      title: "Batman vs Superman is the best movie everr.",
      upvotes: -100,
      downvotes: 0,
    });
	}

  handleChange = (e) => {
    this.setState({
      input: e.target.value
    });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    if (isIllegalPostTitle(this.state.input)) { 
      this.setState({ input: "" });
      return;
    }

    this.state.posts.push({
      title: this.state.input,
      upvotes: 0,
      downvotes: 0,
    });

    this.sortPosts();
    this.setState({ input: "" });
  } 

  handleUpvote = (post) => {
    post.upvotes++;
    this.setState({ posts: sortPosts(this.state.posts) });
  }

  handleDownvote = (post) => {
    post.downvotes++;
    this.setState({ posts: this.state.posts });
  }

  render() {
    let posts = this.state.posts;

    if (!posts) {
      console.log("No posts!");
      return false;
    }

    let _this = this;
    return (
      <div className="Posts">
        { Object.keys(posts).map(function(key) {
            return (
              <Post 
                posts = { posts }
                i = { key }
                handleUpvote = { (post) => _this.handleUpvote(post) }
                handleDownvote = { (post) => _this.handleDownvote(post) }
              />
            );
        })}
      </div>
    );
  }
}

function isIllegalPostTitle(s) {
  let isEmptyString = s === "";
  let isWhitespace = !/\S/.test(s);
  let isLongerThan255 = s.length > 255;

  if (isEmptyString || isWhitespace) {
    return true;
  }

  if (isLongerThan255) {
    alert("Title of post has length limit of 255!");
    return true;
  }

  return false;
}

function sortPosts(posts) {
  let postsClone = posts.slice();
  postsClone.sort(function(a, b) { return b.upvotes - a.upvotes; })
  return postsClone;
}

export default App;