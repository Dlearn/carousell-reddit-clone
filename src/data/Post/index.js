export type Post = {
  title: string,
  hoursAgo: number,
  author: string,
  votes: number,
};
