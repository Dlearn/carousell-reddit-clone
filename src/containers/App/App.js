// src/containers/App/App.js

import React, { Component } from 'react';
import { PageHeader } from 'react-bootstrap';

import AddPost from '../../components/AddPost/AddPost';
import Posts from '../../components/Posts/Posts';

import defaultPosts from './defaultPosts';

class App extends Component {
  constructor() {
    super();

    // Populate the list with some dummy posts.
    this.state = {
      posts: defaultPosts,
    };
  }

  createNewPost = (postTitle) => {
    const posts = this.state.posts.slice();
    posts.push({
      title: postTitle,
      hoursAgo: 0,
      author: 'Dylan',
      votes: 0,
    });

    this.setState({ posts: sortPosts(posts) });
  };

  handleUpvote = (i) => {
    const posts = this.state.posts.slice();
    posts[i].votes += 1;

    this.setState({ posts: sortPosts(posts) });
  };

  handleDownvote = (i) => {
    const posts = this.state.posts.slice();
    posts[i].votes -= 1;

    this.setState({ posts: sortPosts(posts) });
  };

  render() {
    // Filter the top 20 posts in the post array
    const filteredPosts = this.state.posts.slice(0, 20);

    return (
      <div className="container">
        <PageHeader>
          <strong>Reddit Clone</strong>
        </PageHeader>
        <AddPost createNewPost={i => this.createNewPost(i)} />

        <Posts
          posts={filteredPosts}
          handleUpvote={i => this.handleUpvote(i)}
          handleDownvote={i => this.handleDownvote(i)}
        />
      </div>
    );
  }
}

export function sortPosts(posts) {
  const postsClone = posts.slice();
  postsClone.sort((a, b) => b.votes - a.votes);
  return postsClone;
}

export default App;
