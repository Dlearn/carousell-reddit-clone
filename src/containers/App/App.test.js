// src/containers/App/App.test.js

import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App, { sortPosts } from './App';
import Post from './../../components/Post/Post';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('sorts correctly', () => {
  let unsortedPosts = [
    {
      title: "My Little Pony is the best show ever",
      hoursAgo: 10,
      author: "3DSUM",
      votes: 90,
    },
    {
      title: "Why isn't Trump impeached yet?",
      hoursAgo: 90,
      author: "crazymunch",
      votes: 100,
    },
  ];
  let sortedPosts = [
    {
      title: "Why isn't Trump impeached yet?",
      hoursAgo: 90,
      author: "crazymunch",
      votes: 100,
    },
    {
      title: "My Little Pony is the best show ever",
      hoursAgo: 10,
      author: "3DSUM",
      votes: 90,
    },
  ];
	expect(sortPosts(unsortedPosts)).toEqual(sortedPosts);
});

test('App matches snapshot', () => {
  const component = renderer.create(
    <App />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('correctly changes the App state when upvote is pressed', () => {
  const wrapper = mount(
    <App />
  );
  const firstPostVotes = wrapper.find('#votes').at(0);
  
  let currentUpvotesString = firstPostVotes.text().split(" ").pop();
  let currentUpvotes = parseInt(currentUpvotesString, 10);
  expect(currentUpvotes).toBe(100);

  const firstPostUpvoteButton = wrapper.find('#upvote-button').at(0);
  firstPostUpvoteButton.simulate('click');
  currentUpvotesString = firstPostVotes.text().split(" ").pop();
  currentUpvotes = parseInt(currentUpvotesString, 10);
  expect(currentUpvotes).toBe(101);
});

it('correctly changes the App state when downvote is pressed', () => {
  const wrapper = mount(
    <App />
  );
  const firstPostVotes = wrapper.find('#votes').at(0);
  
  let currentUpvotesString = firstPostVotes.text().split(" ").pop();
  let currentUpvotes = parseInt(currentUpvotesString, 10);
  expect(currentUpvotes).toBe(101);

  const firstPostDownvoteButton = wrapper.find('#downvote-button').at(0);
  firstPostDownvoteButton.simulate('click');
  currentUpvotesString = firstPostVotes.text().split(" ").pop();
  currentUpvotes = parseInt(currentUpvotesString, 10);
  expect(currentUpvotes).toBe(100);
});