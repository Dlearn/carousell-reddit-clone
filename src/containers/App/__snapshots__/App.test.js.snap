// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="page-header"
  >
    <h1>
      <strong>
        Reddit Clone
      </strong>
    </h1>
  </div>
  <div>
    <div />
    <form
      className="jumbotron myJumbotron"
      onSubmit={[Function]}
    >
      <div
        className="form-group form-group-lg has-feedback"
      >
        <label
          className="h2 control-label"
          htmlFor="formBasicText"
        >
          + Create New Post
        </label>
        <input
          className="form-control"
          id="formBasicText"
          onChange={[Function]}
          placeholder="Write the title of your new post."
          type="text"
          value=""
        />
      </div>
    </form>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        1
        : 
        Why isn't Trump impeached yet?
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        90
         hours ago by 
        crazymunch
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        100
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        2
        : 
        My Little Pony is the best show ever
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        10
         hours ago by 
        3DSUM
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        90
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        3
        : 
        What is your favorite board game?
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        23
         hours ago by 
        carlstones
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        80
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        4
        : 
        The quality of this subreddit is not good...
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        45
         hours ago by 
        51llahw
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        70
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        5
        : 
        Anybody concerned we only have upvotes in multiples of 10s?
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        58
         hours ago by 
        FeversMirrors
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        60
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        6
        : 
        Marvel or DC?
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        32
         hours ago by 
        madam1
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        50
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        7
        : 
        What are some of your favorite books?
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        95
         hours ago by 
        savvylol
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        40
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        8
        : 
        MEEEEEEEEEMES!!
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        39
         hours ago by 
        Pejkster
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        30
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        9
        : 
        How to build a jetpack?
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        76
         hours ago by 
        insoh2
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        29
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        10
        : 
        My Elsa Cosplay! Upvote!
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        28
         hours ago by 
        erinaceidae
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        28
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        11
        : 
        I just read Tale of two Cities. WOW.
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        2
         hours ago by 
        rotateclockwise
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        27
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        12
        : 
        I miss scooby dooby doo...
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        3
         hours ago by 
        PrestinSowers
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        26
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        13
        : 
        Updoot for love y'all!
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        13
         hours ago by 
        KungFooKangaroo
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        25
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        14
        : 
        If you upvote this, they will see this image when they google.
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        64
         hours ago by 
        Stanzin7
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        24
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        15
        : 
        Stranger things have happened since...
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        23
         hours ago by 
        aaronclements
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        3
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        16
        : 
        Anybody want to make friends?
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        45
         hours ago by 
        ivy_tamwood
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        2
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        17
        : 
        I'm not a good painter, but I made this in my spare time.
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        26
         hours ago by 
        Carp8DM
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        -5
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        18
        : 
        I just donated $10000 to North Korea! Help!
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        73
         hours ago by 
        ross-svh-photo
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        -10
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        19
        : 
        Everyone should watch Avatar: The Last Airbender.
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        18
         hours ago by 
        N3ptun3
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        -30
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
  <div
    className="panel panel-default"
    id={undefined}
  >
    <div
      className="panel-body"
    >
      <h2
        id="title"
      >
        20
        : 
        Manga, to read or not to read?
      </h2>
      <h4
        className="text-muted"
        id="subtitle"
      >
        Submitted 
        16
         hours ago by 
        mjc120
        .
      </h4>
      <br />
      <h4
        id="votes"
      >
        Votes: 
        -50
      </h4>
      <button
        className="btn btn-sm btn-success"
        disabled={false}
        id="upvote-button"
        onClick={[Function]}
        type="button"
      >
        Upvote
      </button>
      <button
        className="btn btn-sm btn-danger"
        disabled={false}
        id="downvote-button"
        onClick={[Function]}
        type="button"
      >
        Downvote
      </button>
    </div>
  </div>
</div>
`;
